backend/app/main.py

```diff
@@ -174,6 +176,34 @@ async def shatter_parent(
     return result
 
 
+@app.patch(
+    "/api/update_assignments/{document_id}/unshatter_parents",
+    response_model=UnShatterResult,
+)
+async def unshatter_parent(
+    document_id: str, data: ZoneAndGEOIDS, session: Session = Depends(get_session)
+):
+    stmt = text(
+        """SELECT *
+        FROM unshatter_parent(:input_document_id, :parent_geoids, :input_zone)"""
+    ).bindparams(
+        bindparam(key="input_document_id", type_=UUIDType),
+        bindparam(key="parent_geoids", type_=ARRAY(String)),
+        bindparam(key="input_zone", type_=INT),
+    )
+    results = session.execute(
+        statement=stmt,
+        params={
+            "input_document_id": document_id,
+            "parent_geoids": data.geoids,
+            "input_zone": data.zone,
+        },
+    )
+    session.commit()
+    result = UnShatterResult(parents=GEOIDS(geoids=data.geoids))
```

Comment:
PP to use `RETURNING` from the SQL query rather than just pass back the input data.